newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -381,21 +401,39 @@ const SpritesList = ({
         multiSelection: true,
         resourceKind: 'image',
       });
-
       resources.forEach(resource => {
         applyResourceDefaults(project, resource);
         project.getResourcesManager().addResource(resource);
-
-        const sprite = new gd.Sprite();
-        sprite.setImageName(resource.getName());
-
-        applyPointsAndMasksToSpriteIfNecessary(sprite);
-
-        onSpriteAdded(sprite); // Call the callback before `addSprite`, as `addSprite` will store a copy of it.
-        direction.addSprite(sprite);
-        sprite.delete();
       });
 
+      if (directionSpritesCountBeforeAdding === 0 && resources.length > 1) {
+        const resourcesByAnimation = groupResourcesByAnimations(resources);
+        if (resourcesByAnimation.size > 1) {
+          addAnimations(resourcesByAnimation);
+        } else {
+          // Use `resourcesByAnimation` because frames are sorted.
+          for (const resources of resourcesByAnimation.values()) {
+            for (const resource of resources) {
+              addAnimationFrame(
+                spriteConfiguration,
+                direction,
+                resource,
+                onSpriteAdded
+              );
+            }
+          }
+        }
+      } else {
+        for (const resource of resources) {

```

Comment:
Question: would it make sense to still apply a logic to order resources?

Reply:
I'm not sure because if there are already some frames in the animation, maybe users want to add frames backward to make a loop.